src/applications/disability-benefits/2346/actions/index.js

```diff
@@ -0,0 +1,59 @@
+import { getVeteranInformation } from '../api';
+import {
+  FETCH_VETERAN_ADDRESS_FAILURE,
+  FETCH_VETERAN_ADDRESS_SUCCESS,
+  FETCH_VETERAN_INFORMATION,
+  FETCH_VETERAN_INFORMATION_FAILURE,
+} from '../constants';
+
+export const fetchVeteranAddressSuccess = data => ({
+  type: FETCH_VETERAN_ADDRESS_SUCCESS,
+  data,
+});
+
+export const fetchVeteranAddressFailure = () => ({
+  type: FETCH_VETERAN_ADDRESS_FAILURE,
+});
+
+export const fetchVeteranInformation = data => ({
+  type: FETCH_VETERAN_INFORMATION,
+  data,
+});
+
+export const fetchVeteranInformationFailure = () => ({
+  type: FETCH_VETERAN_INFORMATION_FAILURE,
+});
+
+export const getVeteranInformationData = () => async dispatch => {
+  try {
+    const data = await getVeteranInformation();
+    const veteranInformation = {
+      veteranFullName: data.formData.veteranFullName,
+      gender: data.formData.gender,
+      dateOfBirth: data.formData.dateOfBirth,
+    };
+    dispatch(fetchVeteranInformation(veteranInformation));
+  } catch (error) {
+    dispatch(
+      fetchVeteranInformationFailure(
+        error,
+        'failed to retrieve data from the api',
+      ),
+    );
+  }
+};
+
+export const getVeteranAddressInfo = () => async dispatch => {
+  try {
+    const data = await getVeteranInformation();
```

Comment:
Why does `getVeteranInformationData` and `getVeteranAddressInfo` use the same `getVeteranInformation` call? Can you combine these?

Reply:
We slicing out certain pieces of state in our actions and our reducers which is a pattern endorsed by redux. Yes they are using the same endpoint, but the point is to only use the part of the state we need in a particular view

Reply:
@mojaray2k I think the point is to reduce API calls.  If you're already calling that API and its already returning ALL the data, it makes no sense to store only some of it and then call the same API again to store other pieces since you already have access to it in the return of the original API call.  A singular call would dry this up a lot.  Good catch @Mottie

Reply:
Ok that has been refactored and there is only one api call